dom-fetch-client/src/main/scala/org/http4s/dom/FetchClient.scala

```diff
@@ -49,27 +49,29 @@ object FetchClient {
 
           Fetch.fetch(req.uri.renderString, init)
         }
+      }.toResource
+      status <- F.fromEither(Status.fromInt(response.status)).toResource
+      body <- Resource.makeCase(response.body.pure[F]) {
+        case (r, Resource.ExitCase.Succeeded) => F.fromPromise(F.delay(r.cancel(null))).void
```

Comment:
Nit: Can we provide a reason other than `null`? I know the scala doc says, 

```scala
  /**
   * The cancel method cancels the stream, signaling a loss of interest in
   * the stream by a consumer. The supplied reason argument will be given to
   * the underlying source, which may or may not use it.
   *
   * @param reason the reason <- actually not what type this is
   * @return a Promise, but not quite sure what it can contain
   */
 ```
 
 but it would be nice to not have the `null`s in the code, even just so future linters don't fuss about it.

Reply:
So the official docs are here which indicate it is an optional argument: https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/cancel

Since it's an optional argument, the idiomatic thing to do here would be use `js.undefined` but unfortunately the facade didn't account for this. Hopefully this will be fixed in a future version of scala-js-dom. OTOH, if you think we should provide a _reason_ (instead of just trying to avoid using null/linting complaints), I'm open to that :)